### PR TITLE
ath25: split up DEVICE_TITLE

### PR DESCRIPTION
**Pull request by mwarning**
Splits up DEVICE_TITLE into DEVICE_VENDOR, DEVICE_MODEL and optional DEVICE_VARIANT.

### CODE REVIEW

File: target/linux/ath25/image/Makefile [hunk 1]
```diff
@@ -73,32 +73,37 @@ endef
 TARGET_DEVICES += generic
```
Comment (adrianschmutzler) [inline target/linux/ath25/image/Makefile]: @aparcar How to deal with a generic device like that, which is nevertheless added to TARGET_DEVICES?

Comment (aparcar) [reply target/linux/ath25/image/Makefile]: I don't know how the `TARGET_DEVICES` works, can you elaborate?

Comment (adrianschmutzler) [reply target/linux/ath25/image/Makefile]: If I understand it correctly, it just means that "generic" is built as an actual device and not just a node with common definitions. So one would have to decide whether it is left with DEVICE_TITLE as a special case or one wants to introduce DEVICE_VENDOR etc. here too.

Comment (aparcar) [reply target/linux/ath25/image/Makefile]: My initial idea was to keep DEVICE_TITLE for such special cases, so I'd be fine if these special cases lack vendor etc.

File: target/linux/ath25/image/Makefile [hunk 2]
```diff
@@ -73,32 +73,37 @@ endef
...
+  DEVICE_VENDOR := Ubiquiti
+  DEVICE_MODEL := XS5
   IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | pad-to 128k | mkfwimage XS5 -v XS5.ar2313
 endef
 TARGET_DEVICES += ubnt5
 
 define Device/np25g
-  DEVICE_TITLE := np25g
+  DEVICE_VENDOR := Compex
+  DEVICE_MODEL := np25g
```
Comment (adrianschmutzler) [inline target/linux/ath25/image/Makefile]: I think this should be capital letters.

File: target/linux/ath25/image/Makefile [hunk 3]
```diff
@@ -73,32 +73,37 @@ endef
...
   KERNEL := kernel-bin | gzip-kernel
   IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | pad-to 128k | mkmylofw np25g
 endef
 #TARGET_DEVICES += np25g
 
 define Device/wpe53g
-  DEVICE_TITLE := wpe53g
+  DEVICE_VENDOR := Compex
+  DEVICE_MODEL := wpe53g
```
Comment (adrianschmutzler) [inline target/linux/ath25/image/Makefile]: I think this should be capital letters.

File: target/linux/ath25/image/Makefile (hunk 1 above)
Comment (adrianschmutzler) [reply target/linux/ath25/image/Makefile]: Okay, so keep it as it is.

File: target/linux/ath25/image/Makefile (hunk 3 above)
Comment (mwarning) [reply target/linux/ath25/image/Makefile]: Thanks for the review!

File: target/linux/ath25/image/Makefile (hunk 1 above)
Comment (mwarning) [reply target/linux/ath25/image/Makefile]: Ok, I will leave it as it is then.

File: target/linux/ath25/image/Makefile (hunk 3 above)
Comment (adrianschmutzler) [reply target/linux/ath25/image/Makefile]: I was referring to the DEVICE_MODEL.

Comment (mwarning) [reply target/linux/ath25/image/Makefile]: Ok, I searched compex and saw that is was written upper case on some sites.

**Review by adschm — APPROVED** [review]
LGTM.

**Comment by dangowrt** [comment]
Cherry-picked onto master, thanks!

**Pull request closed without merging**